streaming/src/main/scala/org/apache/spark/sql/execution/streaming/CarbonStreamingQueryListener.scala

```diff
@@ -33,7 +33,16 @@ class CarbonStreamingQueryListener(spark: SparkSession) extends StreamingQueryLi
   private val cache = new util.HashMap[UUID, ICarbonLock]()
 
   override def onQueryStarted(event: StreamingQueryListener.QueryStartedEvent): Unit = {
-    val qry = spark.streams.get(event.id).asInstanceOf[StreamExecution]
+    val streamQuery = spark.streams.get(event.id)
+    val qry = if (streamQuery.isInstanceOf[StreamExecution]) {
```

Comment:
Isn't there an utility to check spark version?

Reply:
fixed